<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GenresOptionsPage</class>
 <widget class="QWidget" name="GenresOptionsPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>590</width>
    <height>471</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QGroupBox" name="use_genres">
     <property name="title">
      <string>Use genres from MusicBrainz</string>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QCheckBox" name="only_my_genres">
        <property name="text">
         <string>Only use my genres</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="artists_genres">
        <property name="text">
         <string>Fall back on album's artists genres if no genres are found for the release or release group</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="folksonomy_tags">
        <property name="text">
         <string>Use folksonomy tags as genre</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Minimal genre usage:</string>
          </property>
          <property name="buddy">
           <cstring>min_genre_usage</cstring>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="min_genre_usage">
          <property name="suffix">
           <string> %</string>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="label_6">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Maximum number of genres:</string>
          </property>
          <property name="buddy">
           <cstring>min_genre_usage</cstring>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="max_genres">
          <property name="maximum">
           <number>100</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QLabel" name="ignore_genres_4">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>4</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Join multiple genres with:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="join_genres">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>1</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="editable">
           <bool>true</bool>
          </property>
          <item>
           <property name="text">
            <string/>
           </property>
          </item>
          <item>
           <property name="text">
            <string> / </string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>, </string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="label_genres_filter">
        <property name="text">
         <string>Genres or folksonomy tags to include or exclude:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="genres_filter">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Lines not starting with &lt;span style=&quot; font-weight:600;&quot;&gt;-&lt;/span&gt; or &lt;span style=&quot; font-weight:600;&quot;&gt;+&lt;/span&gt; are ignored.&lt;/p&gt;&lt;p&gt;Examples:&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;#comment&lt;br/&gt;!comment&lt;br/&gt;comment&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; text-decoration: underline;&quot;&gt;Strict filtering:&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;-word&lt;/span&gt;: exclude this &lt;span style=&quot; font-style:italic;&quot;&gt;word&lt;br/&gt;&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;+word&lt;/span&gt; : include this &lt;span style=&quot; font-style:italic;&quot;&gt;word&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; text-decoration: underline;&quot;&gt;Wildcard filtering:&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;-*word&lt;/span&gt;: exclude all genres ending with &lt;span style=&quot; font-style:italic;&quot;&gt;word&lt;br/&gt;&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;+word*&lt;/span&gt;: exclude all genres starting with &lt;span style=&quot; font-style:italic;&quot;&gt;word&lt;br/&gt;&lt;/span&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;-w*rd&lt;/span&gt;: exclude all genres starting with &lt;span style=&quot; font-style:italic;&quot;&gt;w&lt;/span&gt; and ending with &lt;span style=&quot; font-style:italic;&quot;&gt;rd&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; text-decoration: underline;&quot;&gt;Regular expressions filtering (Python re syntax):&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;-/^w.rd+/&lt;/span&gt;: exclude genres starting with &lt;span style=&quot; font-style:italic;&quot;&gt;w&lt;/span&gt; followed by any character, then &lt;span style=&quot; font-style:italic;&quot;&gt;r&lt;/span&gt; followed by at least one &lt;span style=&quot; font-style:italic;&quot;&gt;d&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_test_genres_filter">
        <property name="text">
         <string>Test genres or folksonomy tags filter</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="test_genres_filter">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;You can add genres to test filter against, one per line&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_test_genres_filter_error">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>181</width>
       <height>31</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
